### PR TITLE
add script delete_rke2_release_assets

### PR DESCRIPTION
**Pull request by brooksn**
https://github.com/drone-plugins/drone-github-release/issues/88 affects all rke2 pipelines that are triggered by a new tag, and subsequently rerun. 

As a workaround, all assets in a release must be deleted before re-running an rke2 pipeline that was triggered by a tag. delete_rke2_release_assets aims to make that easier. The `-f` option is meant for use inside a drone pipeline, so a build step will not delete files other than artifacts that it actually produced. The `-a` option is meant for running the script manually before re-running a pipeline. Since this is just a temporary workaround, drone integration probably doesn't matter.

Related: https://github.com/brooksn/drone-github-release/commit/08db60a8438db916d8f11d61b5e1f09734172ca4

### CODE REVIEW

File: bin/delete_rke2_release_assets [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        *)
+            usage
+            exit 1
+            ;;
+    esac
+done
+
+setup_tmp
+
+echo $TMP_DIR
```
Comment (briandowns) [inline bin/delete_rke2_release_assets:48]: Does this need to be printed out?

File: bin/delete_rke2_release_assets [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+gh release view "${RELEASE_TAG}" -R rancher/rke2 --json assets -q '.assets.[] | .name' | sort > "${TMP_DIR}/${RELEASE_TAG}-github-assets"
+
+if [ ${DELETE_ALL} ]; then
+    DELETIONS=$(cat "${TMP_DIR}/${RELEASE_TAG}-github-assets")
+else
+    ls $FILES | xargs -n1 basename | sort > "${TMP_DIR}/${RELEASE_TAG}-local-assets"
```
Comment (briandowns) [inline bin/delete_rke2_release_assets:55]: Can you put brackets around the variables that don't have them?

File: bin/delete_rke2_release_assets (hunk 1 above)
Comment (brooksn) [reply bin/delete_rke2_release_assets]: No

File: bin/delete_rke2_release_assets (hunk 2 above)
Comment (brooksn) [reply bin/delete_rke2_release_assets]: Yep, I think I got them all.

File: bin/delete_rke2_release_assets [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            exit 1
+            ;;
+    esac
+done
+
+if [ -z "${RELEASE_TAG}" ]; then
+    echo "error: release tag required"
+    exit 1
+fi
+
```
Comment (briandowns) [inline bin/delete_rke2_release_assets:50]: Can we add the `has_gh` function?

**Comment by briandowns** [comment]
Is it possible to make this generic enough to take a `-r` flag for repo and be able to provide either k3s or rke2?

File: bin/delete_rke2_release_assets [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+if [ -z "${DELETIONS}" ]; then
+    echo "No assets to delete"
+    exit 0
+fi
+
+for i in ${DELETIONS}; do
+    if [ ${DRY_RUN} ]; then
+        echo "gh release delete-asset ${RELEASE_TAG} ${i} --repo rancher/image-build-kubernetes -y"
```
Comment (jossemargt-3pillar) [inline bin/delete_rke2_release_assets:69]: I think repository was meant to be set on `rancher/rke2` or based on [briandowns' comment](https://github.com/rancher/ecm-distro-tools/pull/86#issuecomment-1099678615) what comes from `-r` option.

Same goes to L71

**Review by jossemargt-3pillar — APPROVED** [review]

File: bin/delete_rke2_release_assets (hunk 3 above)
Comment (brooksn) [reply bin/delete_rke2_release_assets]: added has_gh

File: bin/delete_rke2_release_assets (hunk 2 above)
Comment (brooksn) [reply bin/delete_rke2_release_assets]: script now only deletes all files, so all variables have brackets and quotes.

File: bin/delete_rke2_release_assets (hunk 3 above)
Comment (brooksn) [reply bin/delete_rke2_release_assets]: added has_gh function

**Review by briandowns — APPROVED** [review]

**Pull request merged**